### ISSUE TITLE
make distclean is required after invoking autogen.sh

### ISSUE THREAD

**Issue by gnzlbg** [jemalloc/jemalloc#1199]
I can compile on MacOS X without issues in the same directory as the jemalloc source code by doing:

```
cd jemalloc
./autogen.sh
./configure
make
```

However, if I try to compile using `--with-jemalloc-prefix` the build fails:

```
cd jemalloc
./autogen.sh
../configure --with-jemalloc-prefix=_rjem_
make
```

Many of the warnings are of the form:

```shell
In file included from ../src/jemalloc.c:2:
In file included from ../include/jemalloc/internal/jemalloc_preamble.h:43:
include/jemalloc/internal/private_namespace.h:16:9: warning: 'je_aligned_alloc' macro redefined [-Wmacro-redefined]
#define je_aligned_alloc JEMALLOC_N(je_aligned_alloc)
        ^
../include/jemalloc/internal/../jemalloc.h:59:11: note: previous definition is here
#  define je_aligned_alloc je_aligned_alloc
```

and the error message I get is:

```shell
../src/jemalloc.c:3117:1: error: conflicting types for 'je_je_malloc_stats_print'
je_malloc_stats_print(void (*write_cb)(void *, const char *), void *cbopaque,
^
include/jemalloc/internal/private_namespace.h:25:31: note: expanded from macro 'je_malloc_stats_print'
#define je_malloc_stats_print JEMALLOC_N(je_malloc_stats_print)
                              ^
../include/jemalloc/internal/jemalloc_preamble.h:20:25: note: expanded from macro 'JEMALLOC_N'
#  define JEMALLOC_N(n) je_##n
                        ^
<scratch space>:60:1: note: expanded from here
je_je_malloc_stats_print
^
../src/jemalloc.c:661:2: note: previous implicit declaration is here
        je_malloc_stats_print(NULL, NULL, opt_stats_print_opts);
        ^
include/jemalloc/internal/private_namespace.h:25:31: note: expanded from macro 'je_malloc_stats_print'
#define je_malloc_stats_print JEMALLOC_N(je_malloc_stats_print)
                              ^
../include/jemalloc/internal/jemalloc_preamble.h:20:25: note: expanded from macro 'JEMALLOC_N'
#  define JEMALLOC_N(n) je_##n
                        ^
<scratch space>:179:1: note: expanded from here
je_je_malloc_stats_print
^
```

If I run `make distclean` after running `autogen.sh`, the build succeeds:

```
cd jemalloc
./autogen.sh
make distclean
../configure --with-jemalloc-prefix=_rjem_
make
```

Note that the `--with-jemalloc-prefix` option is not covered by CI.

**Comment by interwq** [comment]
The issue was caused by the including order of the paths. Should be fixed by #1211.

**Comment by interwq** [comment]
Certainly. The PR is in the jemalloc-ci repo: https://github.com/jemalloc/jemalloc-ci/pull/12
